sql: zero out crdb_internal.node_statement_statistics(anonymized) column

rafiss opened the pull request:
This column is not used, and is expensive to compute. In the past, we
used to redact identifiers from queries for data privacy reasons, but
now we no longer have this policy. (Constants and user-supplied data are
still redacted.) To mitigate impact, we zero out the column value
instead of removing the column completely.

Epic: None
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/105114)
<!-- Reviewable:end -->

rafiss committed:
sql: zero out crdb_internal.node_statement_statistics(anonymized) column

This column is not used, and is expensive to compute. In the past, we
used to redact identifiers from queries for data privacy reasons, but
now we no longer have this policy. (Constants and user-supplied data are
still redacted.) To mitigate impact, we zero out the column value
instead of removing the column completely.

Release note: None

rafiss commented:
tftr!

bors r=knz

rafiss commented:
fixing a merge conflict

bors r-

rafiss commented:
bors r=knz
